tests/testData.h

```diff
@@ -0,0 +1,555 @@
+    }
+    // Test Data3D (by tags)
+    [[nodiscard]] bool checkData3D(std::string_view tagA, std::string_view tagB, double tolerance = 5.0e-3,
+                                   Error::ErrorType errorType = Error::ErrorType::EuclideanError)
+    {
+        auto optDataA = dissolve_.processingModuleData().search<const Data3D>(tagA);
+        if (!optDataA)
+            throw(std::runtime_error(fmt::format("No data with tag '{}' exists.\n", tagA)));
+
+        auto optDataB = dissolve_.processingModuleData().search<const Data3D>(tagB);
+        if (!optDataB)
+            throw(std::runtime_error(fmt::format("No data with tag '{}' exists.\n", tagB)));
+
+        return checkData3D(optDataA->get(), tagA, optDataB->get(), tagB, tolerance, errorType);
+    }
+    // Test SampledVector data
+    [[nodiscard]] bool checkSampledVector(std::string_view tag, const std::vector<double> &referenceData,
+                                          double tolerance = 5.0e-3,
+                                          Error::ErrorType errorType = Error::ErrorType::EuclideanError)
+    {
+        // Locate the target reference data
+        auto optData = dissolve_.processingModuleData().search<const SampledVector>(tag);
+        if (!optData)
+            throw(std::runtime_error(fmt::format("No data with tag '{}' exists.\n", tag)));
+        const SampledVector &data = optData->get();
+
+        // Generate the error estimate and compare against the threshold value
+        auto error = Error::error(errorType, data.values(), referenceData).error;
+        auto notOK = isnan(error) || error > tolerance;
+        Messenger::print("Target data '{}' has error of {:7.3e} with reference data and is {} (threshold is {:6.3e})\n\n", tag,
+                         error, notOK ? "NOT OK" : "OK", tolerance);
+        return !notOK;
+    }
+    // Test Vec3 vector data
+    void checkVec3Vector(const std::vector<Vec3<double>> &A, const std::vector<Vec3<double>> &B, double tolerance)
+    {
+        ASSERT_EQ(A.size(), B.size());
+        for (auto n = 0; n < A.size(); ++n)
+        {
+            EXPECT_NEAR(A[n].x, B[n].x, tolerance);
+            EXPECT_NEAR(A[n].y, B[n].y, tolerance);
+            EXPECT_NEAR(A[n].z, B[n].z, tolerance);
+        }
+    }
+    // Test Vec3 vector data (by tag and external data)
+    void checkVec3Vector(std::string_view tag, ForceImportFileFormat externalForces, double tolerance)
+    {
+        auto &vec = dissolve_.processingModuleData().value<std::vector<Vec3<double>>>(tag);
+        std::vector<Vec3<double>> B(vec.size());
+        ASSERT_TRUE(externalForces.importData(B));
+        checkVec3Vector(vec, B, tolerance);
+    }
+    // Test species atom type
+    void checkSpeciesAtomType(const Species *sp, int atomIndex, std::string_view atomTypeName)
+    {
+        ASSERT_TRUE(atomIndex >= 0 && atomIndex < sp->nAtoms());
+        auto &spAtom = sp->atom(atomIndex);
+        auto at = spAtom.atomType();
+        ASSERT_TRUE(at);
```

Comment:
I can understand the intention here, but it might be clearer to explicitly compare to `nullptr` - although it's up to you, ultimately!
```suggestion
        ASSERT_TRUE(at != nullptr);
```

Reply:
Our standards document forbids doing this implicit conversion for `int`s, but not for (shared) pointers. I'd be inclined to keep this as-is, personally.